addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -8409,6 +8416,75 @@ registry.VersionControl = SnippetOptionWidget.extend({
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    async updateUI() {
+        // Waiting for the options to be visible so the alert can be correctly
+        // displayed.
+        this.options.wysiwyg.waitForEmptyMutexAction().then(() => {
+            const alertEl = this.$el[0].querySelector("we-alert");
+            const optionSectionEl = this.$el[0].parentElement;
+            if (!alertEl || optionSectionEl.classList.contains("o_we_outdated_block_options")) {
+                return;
+            }
+            // Adding a class to hide the other options, to only have the alert
+            // displayed (so it is placed correctly).
+            optionSectionEl.classList.add("o_we_outdated_block_options");
+        });
+        return this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Called to replace an outdated snippet by its new version.
+     *
+     * @param {Event} ev
+     */
+    async _onReplaceSnippetClick(ev) {
+        // Getting the new block version.
+        const snippet = this.$target[0].dataset.snippet;
+        const snippetTagName = this.$target[0].tagName.toLowerCase();
+        const customizePanelEl = ev.target.closest(".o_we_customize_panel");
+        const newBlockEl = customizePanelEl.parentElement
+            .querySelector(`${snippetTagName}[data-snippet=${snippet}]`).cloneNode(true);
+        // Replacing the block.
+        this.options.wysiwyg.odooEditor.historyPauseSteps();
+        this.$target[0].classList.add("d-none"); // Hiding the block to replace it smoothly.
+        this.$target[0].insertAdjacentElement("beforebegin", newBlockEl);
+        // Initializing the new block as if it was dropped.
+        await this.options.wysiwyg.snippetsMenu.callPostSnippetDrop($(newBlockEl));
+        await new Promise(resolve => {
+            this.trigger_up("remove_snippet",
+                {$snippet: this.$target, onSuccess: resolve, shouldRecordUndo: false}
+            );
+        });
+        this.options.wysiwyg.odooEditor.historyUnpauseSteps();
+        newBlockEl.classList.remove("oe_snippet_body");
+        this.options.wysiwyg.odooEditor.historyStep();
+    },
+    /**
+     * Called to still access the options of an outdated block, despite the
+     * warning.
+     *
+     * @param {Event} ev
+     */
+    _onAccessOptionsClick(ev) {
+        const alertEl = ev.target.parentElement;
+        const optionSectionEl = alertEl.closest(".o_we_outdated_block_options");
+        alertEl.remove();
+        optionSectionEl.classList.remove("o_we_outdated_block_options");
+        // So the alert does not reappear at each start.
+        this.$target[0].isAlertDiscarded = true;
```

Comment:
`start` is only called once normally?

Reply:
Yes, the "real" start is called only once. But when testing, I noticed that every time we dragged and dropped something in the outdated snippet or the outdated snippet itself, it was started again and so the alert we just discarded was reappearing every time, which I found annoying UX wise :grimacing: So I thought blocking it during the time we are editing was a good thing (of course, the next time we reenter edit mode, the alert is displayed again) :slightly_smiling_face:

Reply:
Mmh indeed we still have the thing which destroys all options and restarts them again when we move things. This is still not a very good practice IMO though... maybe a good alternative would be to make a set and store the snippet names that were discarded?

So:
- On top of the file having, `const controlledSnippets = new Set()`
- Each time an user ignores the alert: `controlledSnippets.add(data-snippet of the target)`
- Each time you want to check: `controlledSnippets.has(data-snippet of the target)`

That way you did not alter the `target` element directly and this is also a bit more robust in case `target` is copied/recreated (the DOM would be the same but `target` would not be the same JS element). And I think that we can consider that if the user discarded the alert for a "countdown" snippet on the page, the alert is discarded for all deprecated countdowns for the current edition.

Still, this is nitpicking of course. I don't think there is something extremely wrong with writing on `this.$target[0]` directly... but not a fan.

Reply:
Just a question for this: is it alright if the alert doesn't reappear until we refresh the page then ? Because `controlledSnippets` is not "cleaned" until a real refresh happens :confused: I tried to clean it in a `cleanForSave` or a `destroy` but then it would also be cleaned at each drag and drop and would therefore be pointless :grimacing:

Reply:
Probably do as `clearM2oRpcCache` does :wink: